Automattic/gridicons
# Allow Gridicons library to be imported by component

The goal of this PR is to prepare Gridicons to be imported by component.

To use the external Gridicon at the moment we do: 

    import Gridicon from 'gridicons';
     <Gridicon icon="external" />

After this PR lands, with the new API we could also import the individual component directly:

    import GridiconExternal from 'gridicons/dist/external';
     <GridiconExternal />

By doing so, we're able to ship only the SVGs that the user needs and the size impact of using the Gridicons library in any project would become as minimal as it can be. For example, at the moment, to use the external Gridicon (278 bytes) in any project you'll pull 81K because the `Gridicon` component contains the whole icon set. By importing the individual component, you'll only pull 1,2K.

Although this introduces the ability to import Gridicons individually, it doesn't remove the current way of importing the whole iconset at once.

# How to test

* Delete your `build/` directory, if necessary.
* Run `npm run build`.
  * Verify that a `build/` directory containing untranspiled files exists.
  * Verify that a `dist/` directory containing files transpiled to ES5 and CommonJS module format exists.
* Run `npm pack` and verify that the `gridicons-2.1.3.tgz` contains: 
  * `dist` dir with the untranspiled files
  * `package.json`
  * `CHANGELOG.md`
  * `LICENSE.md`
  * `README.md`
